arangod/Aql/AqlValue.cpp

```diff
@@ -81,302 +83,209 @@ static inline uint8_t intLength(int64_t value) noexcept {
+      return s.isNull() || (emptyIsNull && s.isNone());
     }
     case VPACK_SLICE_POINTER: {
-      // not resolving externals here
-      VPackSlice s(_data.pointerMeta.pointer);
-      return (s.isNull() || (emptyIsNull && s.isNone()));
+      VPackSlice s{_data.slicePointerMeta.pointer};
+      return s.isNull() || (emptyIsNull && s.isNone());
     }
     case VPACK_MANAGED_SLICE: {
-      VPackSlice s(_data.managedSliceMeta.managedPointer);
-      s = s.resolveExternal();
-      return (s.isNull() || (emptyIsNull && s.isNone()));
-    }
-    case RANGE: {
-      break;
+      VPackSlice s{_data.managedSliceMeta.pointer};
+      return s.isNull() || (emptyIsNull && s.isNone());
     }
+    default:
+      return false;
   }
-
-  return false;
 }
 
-/// @brief whether or not the value is a boolean value
 bool AqlValue::isBoolean() const noexcept {
-  switch (type()) {
-    case VPACK_INLINE_INT48:
-    case VPACK_INLINE_INT64:
-    case VPACK_INLINE_UINT64:
-    case VPACK_INLINE_DOUBLE:
+  auto t = type();
+  switch (t) {
+    case VPACK_INLINE:
+      return VPackSlice{_data.inlineSliceMeta.slice}.isBoolean();
+    case VPACK_SLICE_POINTER:
+      return VPackSlice{_data.slicePointerMeta.pointer}.isBoolean();
+    case VPACK_MANAGED_SLICE:
+      return VPackSlice{_data.managedSliceMeta.pointer}.isBoolean();
+    default:
       return false;
-    case VPACK_INLINE: {
-      return VPackSlice(_data.inlineSliceMeta.slice)
-          .resolveExternal()
-          .isBoolean();
-    }
-    case VPACK_SLICE_POINTER: {
-      // not resolving externals here
-      return VPackSlice(_data.pointerMeta.pointer).isBoolean();
-    }
-    case VPACK_MANAGED_SLICE: {
-      return VPackSlice(_data.managedSliceMeta.managedPointer)
-          .resolveExternal()
-          .isBoolean();
-    }
-    case RANGE: {
-      break;
-    }
   }
-
-  return false;
 }
 
-/// @brief whether or not the value is a number
 bool AqlValue::isNumber() const noexcept {
-  switch (type()) {
+  auto t = type();
+  switch (t) {
     case VPACK_INLINE_INT48:
     case VPACK_INLINE_INT64:
     case VPACK_INLINE_UINT64:
     case VPACK_INLINE_DOUBLE:
       return true;
-    case VPACK_INLINE: {
-      return VPackSlice(_data.inlineSliceMeta.slice)
-          .resolveExternal()
-          .isNumber();
-    }
-    case VPACK_SLICE_POINTER: {
-      // not resolving externals here
-      return VPackSlice(_data.pointerMeta.pointer).isNumber();
-    }
-    case VPACK_MANAGED_SLICE: {
-      return VPackSlice(_data.managedSliceMeta.managedPointer)
-          .resolveExternal()
-          .isNumber();
-    }
-    case RANGE: {
-      break;
-    }
+    case VPACK_INLINE:
+      return VPackSlice{_data.inlineSliceMeta.slice}.isNumber();
+    case VPACK_SLICE_POINTER:
+      return VPackSlice{_data.slicePointerMeta.pointer}.isNumber();
+    case VPACK_MANAGED_SLICE:
+      return VPackSlice{_data.managedSliceMeta.pointer}.isNumber();
+    default:
```

Comment:
Is it somehow possible that we ever attempt to store a number in a `VPACK_MANAGED_STRING` AqlValue? Then this case would need to be handled here too.

Reply:
No, it's not, you can see it in ctor